File: CarRentalSystem/CarRentalSystemAspects/MessageHandler.cs
using System;
using System.IO;

namespace sf.systems.rentals.cars
{
    public class MessageHandler : IMessageHandler
    {
        private const string LogFilePath = "log.txt";
        private readonly ErrorHandler errorHandler;

        public MessageHandler(ErrorHandler errorHandler)
        {
            this.errorHandler = errorHandler;
        }

        public void Log(string message)
        {
            try
            {
                using (StreamWriter writer = File.AppendText(LogFilePath))
                {
                    writer.WriteLine($"{DateTime.Now.ToString()} - {message}");
                }
            }
            catch (Exception ex)
            {
                errorHandler.HandleError(ex);
            }
        }

        public void ShowMessage(string message)
        {
            Console.WriteLine(message);
        }

        public void LogPlusMessage(string message)
        {
            Log(message);
            ShowMessage(message);
        }
    }
}

File: CarRentalSystemPython/aspects/MessageHandler.py
import datetime

class MessageHandler:

    def __init__(self, error_handler):
        self.log_file_path = "log.txt"
        self.error_handler = error_handler

    def log(self, message):
        try:
            with open(self.log_file_path, 'a') as f:
                f.write(f"{datetime.datetime.now()} - {message}\n")
        except Exception as ex:
            self.error_handler.handle_error(ex)

    def show_message(self, message):
        print(message)

    def log_plus_message(self, message):
        self.log(message)
        self.show_message(message)
